emanate/observers.py

```diff
@@ -100,21 +105,36 @@ def _process_queue(self):
             self._handle_event(event)
 
     def _handle_event(self, event):
+        is_retry = False
         try:
-            self._process_event(event)
+            while True:
+                published = self._process_event(event, is_retry=is_retry)
+                if self.incessant and not published:
+                    log.info('Will attempt to republish event.')
+                    time.sleep(0.2) # Wait briefly before trying again.
+                    is_retry = True
+                    continue
+                return
         except Exception as exc:
             log.exception(exc)
         finally:
             self._queue.task_done()
 
-    def _process_event(self, event):
+    def _process_event(self, event, is_retry=False):
+        '''Returns true if the event was published, false otherwise.'''
         properties = Properties()
         if event.context:
             properties.headers = event.context
 
         try:
             self.publisher.publish(self.exchange, '', event.data, properties)
+            return True
         except (AMQPError, RecursionError) as exc:
-            log.exception('Cannot publish to RabbitMQ: %r', exc)
-            log.warn('Failed to publish message payload %s with context %s',
-                     event.data, event.context)
+            # To stop us generate too many logs when repeatedly retrying to
+            # publish an event, we only produce a full log on the initial
+            # attempt.
+            if not is_retry:
```

Comment:
I can see the advantage of not logging the event data and context repeatedly.  However, the exception can change from attempt to attempt, and so should probably be logged for each attempt.